src/Sylius/Bundle/ProductBundle/Form/EventSubscriber/GenerateProductVariantsSubscriber.php

```diff
@@ -19,13 +19,22 @@
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Symfony\Component\Form\FormEvent;
 use Symfony\Component\Form\FormEvents;
-use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
 use Webmozart\Assert\Assert;
 
 final class GenerateProductVariantsSubscriber implements EventSubscriberInterface
 {
-    public function __construct(private ProductVariantGeneratorInterface $generator, private Session $session)
+    public function __construct(private ProductVariantGeneratorInterface $generator, private /* Session */ $requestStack)
     {
+        /** @phpstan-ignore-next-line */
+        if (!$requestStack instanceof SessionInterface && !$requestStack instanceof RequestStack) {
+            throw new \InvalidArgumentException(sprintf('The second argument of "%s" should be instance of "%s" or "%s"', __METHOD__, SessionInterface::class, RequestStack::class));
+        }
```

Comment:
Can't we use unions? 
```suggestion
    public function __construct(private ProductVariantGeneratorInterface $generator, private SessionInterface|RequestStack $requestStackOrSession)
    {
```